core/config/configtest/config.go

```diff
@@ -38,15 +43,55 @@ func dirExists(path string) bool {
 // maintained with the source tree. This should only be used in a
 // test/development context.
 func GetDevConfigDir() string {
```

Comment:
I like the structure -- highly readable. 

Just one minor suggestion: 

[In the formatting rules section](https://docs.google.com/document/d/1mU4S4U1dDpBk9AA180RMRB7g3OH887nqbcEPGD3px3Q/edit#), I have specified the following:

1. Closely related concepts should not be kept far within a file; Should never be kept in a different file.
2. If one function calls another, they should be vertically close. The caller should be placed above the callee.
3. Certain bits of code which has a conceptual affinity towards other code, they should be kept together. Affinity might be caused because a group of functions performs similar operations.
4. The most important concept first and the low-level details in the bottom.

While the current code structure satisfies these rules somewhat, it would be nice to have `gomodDevConfigDir()` immediately after the `GetDevConfigDir()` followed by `gopathDevConfigDir()`. The structure can look similar to how various function calls are made in the top-level function that is `GetDevConfigDir()`

When I read the top-level function, I searched for `gomodDevConfigDir` and had to browse the code a bit more to find out where it is located. 

As these points are highly debatable, I will leave the decision to you.

Reply:
I'd say feel free to create a change on top of this if you like @cendhu

Reply:
Yeah, this is simple to do. Will push a PR.